src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -620,7 +622,7 @@ namespace winrt::TerminalApp::implementation
                 _nextPaneId++;
             }
         });
-
+        pane->EnableBroadcast(_tabStatus.IsInputBroadcastActive());

```

Comment:
this is the scary for me - will a pane moved into a different tab continue broadcasting to the old tab? we don't do any event hookups or unhooks here.

Reply:
Nope, works like a charm

Reply:
Huh! Can you walk me through why it works?
As a reviewer I couldn't see the machinery that kept it safe. I'd rather know that we know the code is safe than that we tested and found the behavior safe :smile:

Reply:
Okay, I know this is gonna cause you physical pain to read:


* `TerminalPage::_HandleMovePane`
  * `TerminalPage::_MovePane`
    * `TerminalTab::DetachPane`
      * `Pane::DetachPane`
        * `this._DetachedHandlers` -> handled in a lambda in `TerminalTab::_AttachEventHandlersToPane`
          * revokes all our per-pane event handlers
          * `TerminalTab::_DetachEventHandlersFromControl`
            * revokes per-control handlers (including these)
    * `TerminalTab::AttachPane`
      * reattaches them to the new tab.